crates/starknet/src/state/mod.rs

```diff
@@ -437,6 +635,233 @@ mod tests {
         assert_eq!(state.get_nonce(&address).unwrap(), Felt::from(0));
     }
 
+    #[test]
+    fn check_devnet_state_with_starknet_in_rust_in_memory_state_reader() {
+        let mut in_memory_state_reader = InMemoryStateReader::default();
+        let (devnet_state, class_hash, address, storage_key) = setup_devnet_state();
+
+        devnet_state.address_to_class_hash.iter().for_each(|(k, v)| {
+            in_memory_state_reader
+                .address_to_class_hash
+                .insert(starknet_in_rust::utils::Address::from(*k), v.bytes());
+        });
+
+        devnet_state.address_to_nonce.iter().for_each(|(k, v)| {
+            in_memory_state_reader
+                .address_to_nonce
+                .insert(starknet_in_rust::utils::Address::from(*k), v.into());
+        });
+
+        devnet_state.address_to_storage.iter().for_each(|(k, v)| {
+            in_memory_state_reader.address_to_storage.insert(k.into(), v.into());
+        });
+
+        devnet_state.class_hash_to_compiled_class_hash.iter().for_each(|(k, v)| {
+            in_memory_state_reader.class_hash_to_compiled_class_hash.insert(k.bytes(), v.bytes());
+        });
+
+        devnet_state.class_hash_to_compiled_class.iter().for_each(|(k, v)| {
+            in_memory_state_reader
+                .class_hash_to_compiled_class
+                .insert(k.bytes(), CompiledClass::try_from(v.clone()).unwrap());
+        });
+
+        fn assert_equal_results(
```

Comment:
I feel like we could reduce some line count if we used macros here, both for `fn assert_equal_results` and all the repetitive matches inside the function body. But I don't want to get into that now.

Reply:
me too